dio/test/interceptor_test.dart

```diff
@@ -349,6 +349,40 @@ void main() {
         expect(response.requestOptions.contentType, isNull);
       });
 
+      test("Interceptor gets stacktrace in onError", () async {
+        final dio = Dio();
+        dio.options.baseUrl = EchoAdapter.mockBase;
+        dio.httpClientAdapter = EchoAdapter();
+
+        late final StackTrace caughtStackTrace;
+        dio.interceptors.addAll([
+          InterceptorsWrapper(
+            onError: (err, handler) {
+              caughtStackTrace = err.stackTrace!;
+              handler.next(err);
+            },
+          ),
+          InterceptorsWrapper(onRequest: (options, handler) {
+            final error = DioError(
+              error: Error(),
+              requestOptions: options,
+            );
+            handler.reject(error, true);
+          }),
+        ]);
+
+        await expectLater(
+          dio.get('/error'),
+          throwsA(predicate(
+            (error) {
+              return error is DioError &&
+                  caughtStackTrace.toString() == error.stackTrace.toString();
+            },
+          )),
```

Comment:
`predicate`s seem to be redundant.

```suggestion
          throwsA(
            (e) => e is DioError && caughtStackTrace.toString() == e.stackTrace.toString()
          ),
```